src/components/HabitForm/HabitForm.js

```diff
@@ -0,0 +1,61 @@
+import "./HabitForm.css"
+import { AppContext } from "../../utils/context.js"
+import { QUERY_HABITS } from "../../utils/graph_queries"
+import { useQuery } from "@apollo/client"
+import { useEffect, useContext } from "react"
+import HabitCard from "../HabitCard/HabitCard"
+import { useMutation } from "@apollo/client"
+import { SUBMIT_HABIT } from "../../utils/graph_mutations"
+import "../../assets/icons/habit7-uncheck.png"
+
+const HabitForm = () => {
+  const { loading, error, data } = useQuery(QUERY_HABITS)
+  const { userHabits, setUserHabits, checkedHabitId } = useContext(AppContext)
+  const [createHabitEntry] = useMutation(SUBMIT_HABIT)
+
+  // const [checkedCards, setCheckedCards] = useState([])
+
+  useEffect(() => {
+    if (!loading && data) {
+      setUserHabits(data.fetchHabit)
+    }
+  }, [data])
+
+  // useEffect(() => {
+  //   const checkedHabits = checkedHabitId.map(ele => {
+  //     return userHabits.find(habit => habit.id === ele)
+  //   })
+  //   const checkedHabitCards = checkedHabits.map(habit => <HabitCard name={habit.name} id={habit.id} key={habit.id} />)
+  //   setCheckedCards(checkedHabitCards)
+  // }, [checkedHabitId])
+
+  const habitCards = userHabits.map(habit => <HabitCard name={habit.name} id={habit.id} key={habit.id} />)
+
+  const createHabitEntries = e => {
+    e.preventDefault()
+    const entryParams = checkedHabitId.map(ele => ({ id: ele }))
+    if (!entryParams.length) {
+      alert("You have not selected any habit")
+    } else {
+      createHabitEntry({ variables: { idArr: entryParams } })
+      alert("Great job")
```

Comment:
How could we accomplish error handling and submitting the form without alerts? Maybe conditional rendering?

Reply:
Gonna open an issue for that